// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package teeapps.params;

message DatasetSplitParams {
  // Proportion of the dataset to include in the train subset
  //  range: [0,1]
  //  default: 0.75
  float training_data_ratio = 1;
  // Whether to fix random
  //  false: sklearn.model_selection.train_test_split(random_state=None)
  //  true: sklearn.model_selection.train_test_split(random_state=random_state)
  bool should_fix_random = 2;
  // Specify the random seed of the shuffling
  //  default: 1024
  int64 random_state = 3;
  // Whether to shuffle the data before splitting
  //  default: true
  bool shuffle = 4;
}